pylink/library.py

```diff
@@ -114,12 +114,9 @@ def find_library_windows(cls):
 
                 # Find all the versioned J-Link directories.
                 ds = filter(lambda x: x.startswith('JLink'), os.listdir(dir_path))
-                for jlink_dir in ds:
-                    # The DLL always has the same name, so if it is found, just
-                    # return it.
-                    lib_path = os.path.join(dir_path, jlink_dir, dll)
-                    if os.path.isfile(lib_path):
-                        yield lib_path
+                lib_path = os.path.join(dir_path, ds[-1], dll) # use the latest jlink DLL (ds[-1])
```

Comment:
So this won't necessarily return the newest version.  `os.listdir` returns the directories in the order specified by the underlying filesystem.  In some cases, this is just the latest directory created, which is generally the latest installed version, but can be an older version if someone switches between versions often.

Reply:
I agree... However I am not sure I have the bandwidth and the environment to test it also on Linux and Mac. 
A logic behavior would be to make the user either to pic a specific version and as default, if nothing is specified, use the latest.

Maybe somebody can improve this and provide implementation for Mac and Linux?

Reply:
Yah, I like the idea of that behaviour.  The reason why I said that `listdir` might not be the best approach is that the order isn't guaranteed, so if we say that we always return the latest one, then we have to do some sorting in addition to `listdir`.  To sort properly, we'll need to handle the different file naming conventions that have been had over time.